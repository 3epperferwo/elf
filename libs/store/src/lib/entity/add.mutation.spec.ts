import { createEntitieStore, createTodo } from '../mocks/stores.mock';
import { addEntity } from './add.mutation';

describe('add', () => {
  let store: ReturnType<typeof createEntitieStore>;

  beforeEach(() => {
    store = createEntitieStore();
  });

  it('should add entity', () => {
    store.reduce(addEntity(createTodo(1)));
    expect(store.getValue()).toMatchSnapshot();
  });

  it('should add multiple entities', () => {
    store.reduce(addEntity([createTodo(1), createTodo(2)]));
    expect(store.getValue()).toMatchSnapshot();
  });

  it('should prepend entities', () => {
    store.reduce(addEntity([createTodo(1), createTodo(2)]));
    store.reduce(addEntity([createTodo(3), createTodo(4)], { prepend: true }));
    expect(store.getValue()).toMatchSnapshot();
  });

  it('should overwrite entities', () => {
    store.reduce(addEntity([createTodo(1), createTodo(2)]));
    store.reduce(addEntity([createTodo(3), createTodo(4)], { overwrite: true }));
    expect(store.getValue()).toMatchSnapshot();
  });

});
