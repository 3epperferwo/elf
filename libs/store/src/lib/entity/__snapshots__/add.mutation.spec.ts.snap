// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add should add entity 1`] = `
Object {
  "$entities": Object {
    "1": Object {
      "completed": false,
      "id": 1,
      "title": "todo 1",
    },
  },
  "$ids": Array [
    1,
  ],
}
`;

exports[`add should add multiple entities 1`] = `
Object {
  "$entities": Object {
    "1": Object {
      "completed": false,
      "id": 1,
      "title": "todo 1",
    },
    "2": Object {
      "completed": false,
      "id": 2,
      "title": "todo 2",
    },
  },
  "$ids": Array [
    1,
    2,
  ],
}
`;

exports[`add should overwrite entities 1`] = `
Object {
  "$entities": Object {
    "3": Object {
      "completed": false,
      "id": 3,
      "title": "todo 3",
    },
    "4": Object {
      "completed": false,
      "id": 4,
      "title": "todo 4",
    },
  },
  "$ids": Array [
    3,
    4,
  ],
}
`;

exports[`add should prepend entities 1`] = `
Object {
  "$entities": Object {
    "1": Object {
      "completed": false,
      "id": 1,
      "title": "todo 1",
    },
    "2": Object {
      "completed": false,
      "id": 2,
      "title": "todo 2",
    },
    "3": Object {
      "completed": false,
      "id": 3,
      "title": "todo 3",
    },
    "4": Object {
      "completed": false,
      "id": 4,
      "title": "todo 4",
    },
  },
  "$ids": Array [
    3,
    4,
    1,
    2,
  ],
}
`;
